drainer/server_test.go

```diff
@@ -14,17 +14,286 @@
 package drainer
 
 import (
+	"errors"
 	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
 	"os"
+	"path"
+	"strings"
+	"testing"
+	"time"
 
+	"github.com/gorilla/mux"
 	. "github.com/pingcap/check"
+	pd "github.com/pingcap/pd/client"
+	"github.com/pingcap/tidb-binlog/pkg/etcd"
+	"github.com/pingcap/tidb-binlog/pkg/node"
+	"github.com/pingcap/tidb-binlog/pkg/security"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
 )
 
 var _ = Suite(&testServerSuite{})
 
 type testServerSuite struct{}
 
-func (t *testServerSuite) TestNewServerShouldReturnErr(c *C) {
+func (t *testServerSuite) TestGetLatestTS(c *C) {
+	cp := dummyCheckpoint{commitTS: 1984}
+	server := Server{
+		syncer: &Syncer{
+			cp: &cp,
+		},
+	}
+	router := server.initAPIRouter()
+
+	req := httptest.NewRequest("GET", "/commit_ts", nil)
+	w := httptest.NewRecorder()
+	router.ServeHTTP(w, req)
+
+	resp := w.Result()
+	c.Assert(resp.StatusCode, Equals, http.StatusOK)
+
+	body, _ := ioutil.ReadAll(resp.Body)
+	flatternStr := strings.ReplaceAll(string(body), "\n", " ")
```

Comment:
I didn't do this because:

1. We are not testing the JSON encoding library
1. It would take quite some code to set up JSON decoding in Go.

Does that make sense to you?

Reply:
we test the return format as expect but not JSON encoding library.